### ISSUE TITLE
Locale Support

### ISSUE THREAD

**Issue by ekasprzyk** [contentful-userland/cf-content-types-generator#121]
I came across this tool and it's really nice.  However, I'm a bit confused at the typing on the fields, because as far as I can see, this isn't going to work for anyone, unless we have some special config here.

All our fields from Contentful always have the format:
```
fields: {
  field_name: {
    [LOCALE]: data
  }
}
```
and not
```
fields: {
  field_name: data
}
```
as this library generates.

We have typings for the locales, if you want them.

**Comment by marcolink** [comment]
Hi @ekasprzyk 
The current output of this library only creates non-localized entries. 

To achieve localized entries, I suggest a transformation on type level: [Here](https://www.typescriptlang.org/play?#code/PTAEAsBdIBwZwFwgOYEtLgK4CMB0BjAewFtgiA7SAU0oDNMAbMwymyeh3AKzmGwcLZgxAIZxqAJ2CpyAEyoAPXLNyQ4AYgAyADgCMAKEgBPGFVABRShKMAeAGKoqDWXAB8oALygA3vtCg4I0QfP39QVFkEAMgJGWRQgF9Q2kdnOFAohycXfSTDEzNNQnwRBlQALypZLLT7VJcAGlAikoYqdMVqOXSAayojQlpQEXIjd08QsNAAbRrZADkRYjMZUD6Bobm4AF0AfijfKf9pltKqReXw8mbis5390C3Z+ouqbYBuUP8kvONTG9aFSqlhithB1gAKgUmqc2h0FF0XGt+oNhqNxl5DtMANL9K7IjYWKxGKGmbYZUChXFGUCdGhIgDkKWycAZlP8uwBpSB1XqcBs4JJBRx-W2MNucNcoSigtJVBFRm2uX0+X+AFkjFsJodhlFxLFyPF-NgouRMMRsFQJMrVWYNbCeYKJg7KrJBQLiTYNVtXE0GTQALQAVQAymyAD6gBnyAMAEXMDNcn30FHEoGIRkFMs93r57kxoUCwR1-giUQARLJaHAq+XEg1knyDl9daBy+AnAJyw2pibQAZvjbU5B00YXcDiVF7RLHcTtYWghSS+FIlGMwGIgz642Wc2piI90dK1Q4+Zy1EGcgraJrjERnAGCJIKgWAye0fA6Hz1GaMgynB8HAUA73IB8nxfcg2RbJJBySIA) is a little example how it would look like.

```typescript
type LocalizedFields<Fields, Locales extends keyof any>  = {
    [FieldName in keyof Fields]?: {
      [LocaleName in Locales]?: Fields[FieldName];
    }
}

type LocalizedEntry<EntryType, Locales extends keyof any>  = {
 [Key in keyof EntryType] : 
  Key extends 'fields' 
  ? LocalizedFields<EntryType[Key], Locales>
  : EntryType[Key]
}

type MyFields = {
  a: string
  b: number
}

type MyEntry = Entry<MyFields>

type MyLocale =  'en-US' | 'de-DE';

type MyLocalizedEntry = LocalizedEntry<MyEntry, MyLocale>;

const myEntry: MyEntry = {
  sys: {
    id: "my-id"
  },
  fields: {
    a: "hello",
    b: 1
  }
}

const myLocalizedEntry: MyLocalizedEntry = {
  sys : {
    id: 'my-id'
  },
  fields: {
    a: {
      "de-DE": 'german translation',
      "en-US": 'english translation' 
    }
  }
}

```
In the example, `Entry` refers to the type definition from [contentful.js](https://github.com/contentful/contentful.js/blob/master/index.d.ts#L81). Please let me know if this works for you!

**Comment by ekasprzyk** [comment]
Hello!

I really wanted to not forget to say, many thanks for this, and it works!  I utterly didn't realise at the time that the management API adds in the locales and the delivery API doesn't (depite having worked with both, this utterly escaped me), and I managed to level up my `keyof` understanding so I actually understand what you're doing and how great and simple an idea it is.  

For anyone else looking to work with the management API, this is a working solution.

   Elizabeth

P.S.  Closing the issue as fixed, but might be worth adding to any documentation you have, so other don't get tripped up!

P.P.S.happy to update the docs with a PR if you don't mind.
